src/include/txn.i

```diff
@@ -34,6 +34,50 @@ __wt_ref_cas_state_int(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t old_state
     return (cas_result);
 }
 
+/*
+ * __wt_txn_context_prepare_check --
+ *     Return an error if the current transaction is in the prepare state.
+ */
+static inline int
+__wt_txn_context_prepare_check(WT_SESSION_IMPL *session)
+{
+    if (F_ISSET(&session->txn, WT_TXN_PREPARE))
+        WT_RET_MSG(session, EINVAL, "not permitted in a prepared transaction");
+    return (0);
+}
+
+/*
+ * __wt_txn_context_check --
+ *     Complain if a transaction is/isn't running.
+ */
+static inline int
+__wt_txn_context_check(WT_SESSION_IMPL *session, bool requires_txn)
+{
+    if (requires_txn && !F_ISSET(&session->txn, WT_TXN_RUNNING))
+        WT_RET_MSG(session, EINVAL, "%s: only permitted in a running transaction", session->name);
+    if (!requires_txn && F_ISSET(&session->txn, WT_TXN_RUNNING))
+        WT_RET_MSG(session, EINVAL, "%s: not permitted in a running transaction", session->name);
+    return (0);
+}
```

Comment:
I assume the two functions above were just code movements. If you made content changes to them, please let me know so that I can closely look.

Reply:
There was a minor change to one function (you don't have to print out `session.name` explicitly, the error functions will do it for you), but otherwise it was just a movement.